nsist: use "/" when comparing fnmatch patterns

davvid opened the pull request:
Canonicalize paths to their Unix representation when comparing fnmatch
patterns.  This makes patterns work across platforms, and
simplifies usage by providing a consistent interface to the
exclude patterns.
    
Closes #39

davvid commented:
Note: this is based off of the same branch as #147 so please merge that one first.

takluyver commented:
Thanks! I like the idea of making it platform independent, but let's normalise the exclude patterns as well - that way the change shouldn't break things for anyone with Windows style paths in their exclude patterns.

Also, it's probably a good idea to note in the docs that the patterns can use either kind of slash, and it will be normalised to `/`.

davvid commented:
Good idea.  I've updated this branch so that exclude patterns are normalized too, and added a note to the docs about the patterns being cross-platform.

takluyver commented on doc/cfgfile.rst:
```diff
@@ -276,6 +276,10 @@ the line with the key:
      will be included (you can not exclude a full package/extra directory
      or a single file listed in ``files``).
+   * Exclude patterns are applied uniformly across platforms and can use
+     either Unix-style forward-slash (`/`), or Windows-style back-slash (`\`)
```
N.B. inline code in rst uses double backticks. It's possible to configure single backticks to mean the same,  but I don't think I have on this project.

We should also build the docs and check whether the backslash shows up - it might be treated as an escape.

takluyver commented:
I'll merge this and fiddle with the docs myself - thanks!
